server/testutils/factories/outOfServiceBed.ts

```diff
@@ -1,22 +1,36 @@
 import { Factory } from 'fishery'
 import { faker } from '@faker-js/faker/locale/en_GB'
+import {
+  NamedId,
+  NewCas1OutOfServiceBed as NewOutOfServiceBed,
+  Cas1OutOfServiceBed as OutOfServiceBed,
+  Cas1OutOfServiceBedCancellation as OutOfServiceBedCancellation,
+} from '../../@types/shared'
 
 import referenceDataFactory from './referenceData'
 import { DateFormats } from '../../utils/dateUtils'
-import { NewOutOfServiceBed, OutOfServiceBed, OutOfServiceBedCancellation } from '../../@types/ui'
 
 export const outOfServiceBedFactory = Factory.define<OutOfServiceBed>(() => ({
   id: faker.string.uuid(),
-  bedId: faker.string.uuid(),
-  bedName: faker.lorem.words(3),
-  roomName: faker.lorem.words(3),
-  notes: faker.lorem.sentence(),
-  startDate: DateFormats.dateObjToIsoDate(faker.date.soon()),
-  endDate: DateFormats.dateObjToIsoDate(faker.date.future()),
-  referenceNumber: faker.string.uuid(),
-  reason: referenceDataFactory.lostBedReasons().build(),
-  status: 'active',
-  cancellation: { id: faker.string.uuid(), createdAt: DateFormats.dateObjToIsoDateTime(faker.date.past()) },
+  createdAt: DateFormats.dateObjToIsoDateTime(faker.date.past()),
+  outOfServiceFrom: DateFormats.dateObjToIsoDate(faker.date.future()),
+  outOfServiceTo: DateFormats.dateObjToIsoDate(faker.date.future()),
+  bed: namedIdFactory.build(),
+  room: namedIdFactory.build(),
+  premises: namedIdFactory.build(),
+  apArea: namedIdFactory.build(),
+  reason: referenceDataFactory.outOfServiceBedReason().build(),
+  referenceNumber: faker.helpers.arrayElement([faker.string.uuid(), undefined]),
+  notes: faker.helpers.arrayElement([faker.lorem.sentence(), undefined]),
+  daysLostCount: faker.number.int(),
+  temporality: faker.helpers.arrayElement(['past', 'current', 'future'] as const),
+  status: faker.helpers.arrayElement(['active', 'cancelled'] as const),
+  cancellation: undefined,
+}))
+
+const namedIdFactory = Factory.define<NamedId>(() => ({
```

Comment:
minor but is this supposed to be `NameId`?

Reply:
Nah I think `NamedId` is correct IE an ID with a name. Is that right @danlivings-dxw ?

Reply:
oh, I didn't realise this was an API type. still a bit puzzling IMO, but not a big deal

Reply:
`NamedId` is correct - the thinking behind why it's called that is that it's an ID of something that was named, rather than it being explicitly a name-ID pair, even though it's represented by one.

Reply:
I like it !